src/main/java/org/spongepowered/common/data/persistence/ConfigurateTranslator.java

```diff
@@ -75,64 +73,56 @@ private static DataContainer translateFromNode(ConfigurationNode node) {
         return dataContainer;
     }
 
-    private void translate(Map<Object, ? extends ConfigurationNode> node, DataView container, Map<?, ?> value) {
+    private void translate(Map<Object, ? extends ConfigurationNode> node, DataView container, Object key, Map<?, ?> value) {
         for (Map.Entry<?, ?> o : value.entrySet()) {
-            translate(node.get(o.getKey()), container);
+            ConfigurationNode n = node.get(o.getKey());
+            if (!n.getValueType().canHaveChildren()) {
+                container.set(of(String.valueOf(o.getKey())), n.getValue());
+            } else if (n.hasListChildren()) {
+                translate(n, container);
+            } else {
+                DataContainer sub = DataContainer.createNew(DataView.SafetyMode.NO_DATA_CLONED);
+                translate(n, sub);
+                container.set(of(String.valueOf(o.getKey())), sub);
+            }
         }
     }
 
-    private List<Object> translate(List<? extends ConfigurationNode> node, DataView container, List<?> value) {
+    private void translate(List<? extends ConfigurationNode> node, DataView container, Object key, List<?> value) {
         List<Object> list = new ArrayList<>(value.size());
         for (int i = 0; i < value.size(); i++) {
```

Comment:
This isn't affected by your change, but this for loop does not need an index - a simple foreach works.